superdesk/internal_destinations.py

```diff
@@ -63,9 +63,12 @@ def handle_item_published(sender, item, desk=None, **extra):
                 continue
             if not filters_service.does_match(content_filter, item):
                 continue
-
-        if dest.get("send_after_schedule", False) and item.get("state") != "published":
-            # if send_after_schedule is set to True and item state is other than published
+        if not extra.get("after_scheduled") and dest.get("send_after_schedule"):
+            # if after_schedule is set to False and send_after_schedule is True
+            # then don't execute
+            continue
+        if extra.get("after_scheduled") and not dest.get("send_after_schedule"):
+            # if after_schedule is set to True and send_after_schedule is False
```

Comment:
I think you have to also handle the case when item is not scheduled and it's called with `after_scheduled=False` , otherwise it would create 2 items imo

Reply:
do you mean that the article is published? Not scheduled; if so, do we want to avoid creating a duplicate item through internal destination ?
and IMO it is called with `after_scheduled=False` only when item is published.

Reply:
yep if it's just published and not scheduled, how many items do you get at the end?

Reply:
one item on the desk where it will be send through internal_destination and one item on the desk where it will be published.

Reply:
I see, ok then. can you pls add some test so we have this fix covered

Reply:
Okay I will